lib_test/client_server_test.ml

```diff
@@ -0,0 +1,75 @@
+(** This module tests the NBD server and the NBD client provided by the core
+    NBD library, by connecting their inputs and outputs together so that they
+    communicate with each other through an in-memory pipe.
+    The server and client IO is run in concurrently by Lwt, in the same
+    process. *)
+
+open Lwt.Infix
+
+let with_channels f =
+  let section = Lwt_log_core.Section.make("with_channels") in
+  let make_channel name (ic, oc) =
+    let read c =
+      Lwt_log.debug_f ~section "%s read: %d" name (Cstruct.len c) >>= fun () ->
+      let len = Cstruct.len c in
```

Comment:
If you move it the line above you can avoid duplivating the call. Similarly in write c